TDD_BMS_CurrentRange.cpp

```diff
@@ -0,0 +1,31 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+
+#include "test/catch.hpp"
+#include "BMS_CurrentRange.h"
+
+
+TEST_CASE("To check empty BMS Current readings range") {
+    int BMS_Current_Range[] = {};
+       int numofReadings = sizeof(BMS_Current_Range) / sizeof(BMS_Current_Range[0]);
+  REQUIRE(BMS_CurrentReading_Validity(numofReadings,BMS_Current_Range) == Invalid_Range);
+}
+
+TEST_CASE("To check valid BMS Current readings range") {
+    int BMS_Current_Range[] = {1,2,3,4,5};
+       int numofReadings = sizeof(BMS_Current_Range) / sizeof(BMS_Current_Range[0]);
+  REQUIRE(BMS_CurrentReading_Validity(numofReadings,BMS_Current_Range) == Valid_Range);
+}
+
+TEST_CASE("To check total number of readings in given range") {
+    int BMS_Current_Range[] = {1,2,4,4,5,8,9,10};
+    int numofReadings = sizeof(BMS_Current_Range) / sizeof(BMS_Current_Range[0]);
+    int defined_range[] ={1,4};
+  REQUIRE(BMS_NumofCurrentReadings_InRange(numofReadings,BMS_Current_Range,defined_range) == 4);
+}
+
+TEST_CASE("To check total number of Continous readings(with difference <=1) in given range") {
```

Comment:
Test cases that use the word 'check' are usually not specific in their description.
E.g., over here the description doesn't say whether the range is inclusive of the limits. To know this, we need to see the code with the <= and >=.
Make the tests as a statement of behavior to pin down any assumptions